src/widget/wcoverart.cpp

```diff
@@ -190,7 +191,11 @@ QPixmap WCoverArt::scaledCoverArt(const QPixmap& normal) {
     if (normal.isNull()) {
         return QPixmap();
     }
-    return normal.scaled(size(), Qt::KeepAspectRatio, Qt::SmoothTransformation);
+    QPixmap scaled;
+    scaled = normal.scaled(size() * getDevicePixelRatioF(this),
```

Comment:
Storing the `devicePixelRatio` in a temporary variable might improve readability. This pattern repeats multiple times.

Maybe it is even possible to extract a function `scalePixmapForDevice(...)` to avoid repeating the whole scaling code again and again. Just in case that we need to adjust it.

Reply:
`getDevicePixelRatioF(this)` is only called twice here. I don't think making a new variable for it would be very helpful. Also recall that is a hack for backwards compatibility with Qt < 5.6. It will be able to be shortened to just `devicePixelRatioF()`.

This already is a function for scaling the cover art. The same code is also used by WSpinny, but I don't think it's worth the complexity to create some way for them to share code that is repeated only twice.